force-app/main/default/classes/AccountAutoDeletionSettingsVMapper.cls

```diff
@@ -0,0 +1,116 @@
+     * @description Empty constructor is not public for Singleton Pattern
+     **********************************************************************************/
+    protected AccountAutoDeletionSettingsVMapper() {
+    }
+
+    /**********************************************************************************
+     * @description Static method to get the current instance for Singleton pattern
+     * @return The instance of AccountAutoDeletionSettingsVMapper.
+     **********************************************************************************/
+    public static AccountAutoDeletionSettingsVMapper getInstance() {
+        if (instance == null) {
+            instance = new AccountAutoDeletionSettingsVMapper();
+        }
+
+        return instance;
+    }
+
+    /*******************************************************************************************************
+     * @description Retrieve a DualListboxVModel representing the Account Record Types currently selected
+     * for deletion when not associated with Contact records and any remaining active account record types
+     * availabe for selection.
+     * @return  A DualListboxVModel
+     *******************************************************************************************************/
+    public virtual DualListboxVModel getAccountAutoDeletionSettingsViewModel() {
+        // get current values set for deletion from hierarchy settings
+        AccountRecordTypeAutoDeletionModel currentAutoDeletionSettings = this.locateAccountRecordTypeAutoDeletionService()
+            .getAccountRecordTypeAutoDeletionModel();
+
+        // get all active account record types in org
+        Map<Id, RecordTypeModel> activeAccountRecordTypeModelsById = this.locateRecordTypeService()
+            .getActiveRecordTypeModelsById(
+                this.locateRecordTypeService().getModelByIdMapForSObjectType(Account.SObjectType).values()
+            );
+
+        // create DualListboxValueVModels for each of the selected and available options
+        List<DualListboxValueVModel> optionsDualListboxValuesList = new List<DualListboxValueVModel>();
+
+        // populate available listbox - should contain all values regardless of if they are selected
+        for (Id recordTypeId : activeAccountRecordTypeModelsById.keySet()) {
+            optionsDualListboxValuesList.add(
+                new DualListboxValueVModel(
+                    activeAccountRecordTypeModelsById.get(recordTypeId).name,
+                    activeAccountRecordTypeModelsById.get(recordTypeId).id
+                )
+            );
+        }
+
+        // populate selected listbox - only requires the String representing the value of the selected option
+        List<String> selectedOptionsDualListboxValuesList = (List<String>) this.locateRecordTypeService()
+            .filterListForValidIds(
+                (List<Id>) currentAutoDeletionSettings.accountRecordTypeIdsForDeletionWhenOrphanedList,
```

Comment:
Unfortunately, we can't guarantee that this is a list of ids. Let's pass in the string list and do a two step process in the actual method that skips a string if it is not an id and then skips if it is not in the set.

When I go and mangle the setting with this implementation it doesn't show up in the UI